### PR TITLE
#163426499 Implement bookmark article functionality

### PR DESCRIPTION
**Pull request by omenkish**
#### What does this PR do?
It implements the functionality that enables a user to bookmark an article. 

#### How should this be manually tested?
Clone the Repo, run npm install, then run npm test.

#### What are the relevant pivotal tracker stories?
[#163426499](https://www.pivotaltracker.com/story/show/163426499)
#### Screenshots (if appropriate)
Screen Shot 2019-03-23 at 10 13 06 AM

### CODE REVIEW

**Review by EfeAgare — DISMISSED** [review]

**Review by ozone4real — DISMISSED** [review]
LGTM

**Review by bobbylemm — DISMISSED** [review]

File: src/redux/actions/bookmarkActions.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      return dispatch({ type: ADD_BOOKMARK, payload: response.bookmark });
+    }
+    toast.success(response.message);
+    return dispatch({ type: REMOVE_BOOKMARK, id });
+  } catch (error) {
+    switch (error.response.status) {
+      case 404:
+        return toast.error(error.response.data.message);
+      case 401:
+        return toast.error(error.response.data.message);
```
Comment (p-funky) [inline src/redux/actions/bookmarkActions.js:18]: Keep things DRY
```suggestion
      case 404:
      case 401:
        return toast.error(error.response.data.message);
```

Comment (p-funky) [reply src/redux/actions/bookmarkActions.js:18]: You can cater for both 404 and 401 with one return.

Comment (omenkish) [reply src/redux/actions/bookmarkActions.js:18]: Okay

**Review by EfeAgare — APPROVED** [review]

**Review by p-funky — APPROVED** [review]

**Review by ozone4real — APPROVED** [review]
LGTM

**Pull request merged**